4.3-pipelines/demonstrators/4.3.1-genotyping/stepB-integrate-results.nf

```diff
@@ -0,0 +1,146 @@
+#!/usr/bin/env nextflow
+
+
+input_data_ch = Channel.fromPath(params.inputData)
+    .splitCsv(header: true, sep: "\t", strip: true)
+    .map { row -> [row.dataset_id, row.input_vcf, row.chr_name_mapping, row.liftover_chain] }
+
+input_dir = file(params.inputData).getParent()
+final_vcf_name = file(params.outputVcf).getName()
+final_vcf_dir = file(params.outputVcf).getParent()
+bin_dir = file(params.binDir)
+
+
+process fetchTargetReferenceGenome {
+
+    errorStrategy "retry"
+    maxRetries 3
+    publishDir params.debugDir
+
+    output:
+        path "reference.fa" into reference_genome_fa
+
+    """
+    wget -qO- ${params.targetReferenceGenomeLink} | gzip -cd > reference.fa
+    """
+
+}
+
+
+process renameChromosomes {
+
+    publishDir params.debugDir
+
+    input:
+        set dataset_id, input_vcf, chr_name_mapping, liftover_chain from input_data_ch
+    output:
+        set dataset_id, file("${dataset_id}.1-renamed.vcf.gz"), liftover_chain into renamed_data_ch
+
+    script:
+
+        if ( chr_name_mapping == "-" )
+            """
+            cp ${input_dir}/${input_vcf} "${dataset_id}.1-renamed.vcf.gz"
```

Comment:
I know you're working with small file but I believe a soft/hard link would work just as well.

Reply:
I was worried a bit about how Nextflow will handle this (because it's also using symlinks under the hood to pass files between processes), but I've tried it out and apparently it works just fine, so I've replaced both instances of `cp` with `ln -S`. Thank you for the suggestion!